### ISSUE TITLE
ApdWinsockPort  Port.Open=true after wsClient fails to connect

### ISSUE THREAD

**Issue by sbridger** [TurboPack/AsyncPro#5]
When using the ApdWinsockPort,
Port.Open remains true after a failed attempt to open a wsClient connection. (eg if there was no server to connect to)

Not sure if there is another property to check?

**Comment by sbridger** [comment]
Looks like it probably does clear itself, just rather later.

**Comment by romankassebaum** [comment]
You are writing so many issues. Since you know this code better than me what about fixing it yourself? You can send your patches and I will merge them to the repo.

**Comment by sbridger** [comment]
I would like to, and will if I can, but:

1) I really have no confidence that what hacks I make to get something working are the correct thing to do, at all. I find the code quite incomprehensible, and mostly just change things until I get a result. I don't know if anybody else really wants my bodges messing up functionality because I don't understand it. If there was someone who understood what was going on in these libraries to comment on changes it would help a lot.

2) I am working with XE2, and using my own hacked version of apro. I will try and sync with your version when I move Realterm to a current version, but D keeps moving faster than I catch up, so not sure i is going to happen.

**Comment by romankassebaum** [comment]
The code is very old and I don't use it in a single real project. I can only have a deep look at it. But anyway I can create a XE2 branch for you and maybe you can make it in the first step compilable. Then you don't need to use a hack version.

**Comment by sbridger** [comment]
Yes old indeed. I just put the paper manuals for the TurboPascal version of Apro in the fire last week (winter is coming here), so it is even older than you thought...
Realterm is probably the biggest active application of Apro given the [downloads](https://sourceforge.net/projects/realterm/files/stats/timeline?dates=2002-11-16+to+2020-05-31)

No need to branch now, I will tell you when this gets up the to-do list high enough for me to start. (I also need to get GIT going again, and also try the new github to sourceforge linking)

BTW, it would be good to have enough permissions so I can mark my issues as bugs/feature requests and so on.

I should get on to the EMB people and see if there is a subscription deal that would be workable on the donations for free software projects.

**Comment by romankassebaum** [comment]
Where are you located? Here in Europe summer is coming.

I added you as a reader to the team. If you want to checkin then please tell me. I will add you to the writers.

**Comment by sbridger** [comment]
New Zealand. Been bit grey for a May, my hot water is down to 40C. Need more sun.

**Comment by romankassebaum** [comment]
Northern or Southern Island?

**Comment by sbridger** [comment]
NI - Auckland

**Comment by sbridger** [comment]
I still don't seem to be able to set labels anywhere, maybe I need to be a writer for this...

**Comment by romankassebaum** [comment]
I added you to the writers.
